engine/runtime-integration-tests/src/test/java/org/enso/interpreter/runtime/ManagedResourceTest.java

```diff
@@ -0,0 +1,116 @@
+package org.enso.interpreter.runtime;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.lang.ref.Reference;
+import java.lang.ref.WeakReference;
+import org.enso.common.MethodNames;
+import org.enso.test.utils.ContextUtils;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Source;
+import org.graalvm.polyglot.Value;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ManagedResourceTest {
+  private static Context ctx;
+  private static EnsoContext ensoCtx;
+  private static Value newRef;
+  private static Value createRef;
+  private static Value getRef;
+  private static Value finalizeRef;
+
+  @BeforeClass
+  public static void initCtx() throws Exception {
+    ctx = ContextUtils.createDefaultContext();
+    ensoCtx = ContextUtils.leakContext(ctx);
+    var code =
+        """
+              import Standard.Base.Runtime.Managed_Resource.Managed_Resource
+
+              new_ref obj =
+                Managed_Resource.register obj (_->0)
+
+              create_ref obj system_resource =
+                Managed_Resource.register obj (_->0) system_resource
+
+              get_ref ref = ref.with it->
+                it
+              finalize_ref ref = ref.finalize
+              """;
+    var src = Source.newBuilder("enso", code, "gc.enso").build();
+    var gcEnso = ctx.eval(src);
+    newRef = gcEnso.invokeMember(MethodNames.Module.EVAL_EXPRESSION, "new_ref");
+    createRef = gcEnso.invokeMember(MethodNames.Module.EVAL_EXPRESSION, "create_ref");
+    getRef = gcEnso.invokeMember(MethodNames.Module.EVAL_EXPRESSION, "get_ref");
+    finalizeRef = gcEnso.invokeMember(MethodNames.Module.EVAL_EXPRESSION, "finalize_ref");
+  }
+
+  @AfterClass
+  public static void closeCtx() throws Exception {
+    ctx.close();
+    ctx = null;
+  }
+
+  @Test
+  public void regularReference() throws Exception {
```

Comment:
After copy-pasting from `RefTest.java`, please, at least rename names of the tests, as well as the private static fields like `newRef`, `createRef`, etc.

Reply:
Renamed in 873bac3